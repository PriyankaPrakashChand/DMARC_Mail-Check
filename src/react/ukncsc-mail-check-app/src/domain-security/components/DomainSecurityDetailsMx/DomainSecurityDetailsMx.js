import React from 'react';
import PropTypes from 'prop-types';
import { Divider, Message } from 'semantic-ui-react';
import moment from 'moment';
import { Breadcrumb, BreadcrumbItem } from 'common/components';
import {
  DomainSecurityTitle,
  DomainSecurityDetailsMessages,
} from 'domain-security/components';
import { DomainSecurityLocationContext } from 'domain-security/context';

const DomainSecurityDetailsMx = ({
  domainId,
  mxId,
  hostname,
  domainName,
  type,
  loading,
  error,
  records,
  children,
}) => {
  const record = mxId
    ? records.find(r => String(r.id) === mxId)
    : records.find(r => String(r.hostname) === hostname);

  const recordPrefix = type === 'DKIM' ? 'Selector : ' : '';

  return (
    <React.Fragment>
      <Breadcrumb>
        <DomainSecurityLocationContext.Consumer>
          {location => (
            <BreadcrumbItem link={`/${location}/${domainId}`}>
              {type === 'TLS' ? hostname : domainName}
            </BreadcrumbItem>
          )}
        </DomainSecurityLocationContext.Consumer>
        <BreadcrumbItem active>{type}</BreadcrumbItem>
      </Breadcrumb>
      <DomainSecurityTitle
        title={type}
        subtitle={record && `${recordPrefix}${record.hostname}`}
        loading={loading}
        failures={record && record.failures}
        warnings={record && record.warnings}
        inconclusives={record && record.inconclusives}
      >
        {record &&
          record.lastChecked && (
            <p>Last checked {moment.utc(record.lastChecked).local().fromNow()}</p>
          )}
      </DomainSecurityTitle>
      <Divider hidden />
      {children}
      {error && <Message error>{error.message}</Message>}
      {!loading &&
        !error &&
        record && (
          <DomainSecurityDetailsMessages
            type={type}
            failures={record.failures}
            warnings={record.warnings}
            inconclusives={record.inconclusives}
          />
        )}
    </React.Fragment>
  );
};

DomainSecurityDetailsMx.defaultProps = {
  mxId: null,
  hostname: null,
  loading: false,
  error: null,
  records: [],
};

DomainSecurityDetailsMx.propTypes = {
  mxId: PropTypes.string,
  hostname: PropTypes.string,
  type: PropTypes.string.isRequired,
  loading: PropTypes.bool,
  error: PropTypes.shape({ message: PropTypes.string }),
  records: PropTypes.arrayOf(
    PropTypes.shape({
      hostname: PropTypes.string.isRequired,
      lastChecked: PropTypes.string.isRequired,
      failures: PropTypes.arrayOf(PropTypes.string),
      warnings: PropTypes.arrayOf(PropTypes.string),
      inconclusives: PropTypes.arrayOf(PropTypes.string),
    })
  ),
};

export default DomainSecurityDetailsMx;
