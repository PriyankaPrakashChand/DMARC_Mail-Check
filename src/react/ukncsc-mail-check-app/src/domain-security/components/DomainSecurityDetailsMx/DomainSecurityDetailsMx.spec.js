import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render } from 'react-testing-library';
import DomainSecurityDetailsMx from './DomainSecurityDetailsMx';

describe('DomainSecurityDetailsMx', () => {
  let container;

  describe('when no records are provided', () => {
    beforeEach(() => {
      ({ container } = render(
        <MemoryRouter>
          <DomainSecurityDetailsMx domainId="123" type="SPF">
            <p>bleh</p>
          </DomainSecurityDetailsMx>
        </MemoryRouter>
      ));
    });

    test('it should match the snapshot', () =>
      expect(container).toMatchSnapshot());

    test('it should not display a record', () =>
      expect(
        container.getElementsByClassName('DomainSecurityRecord')
      ).toHaveLength(0));
  });
});
