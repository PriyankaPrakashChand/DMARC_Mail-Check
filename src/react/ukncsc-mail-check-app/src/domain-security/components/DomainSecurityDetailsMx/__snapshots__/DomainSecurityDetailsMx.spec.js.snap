// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DomainSecurityDetailsMx when no records are provided it should match the snapshot 1`] = `
<div>
  <div>
    <div
      class="ui breadcrumb"
    >
      <div
        class="section"
      >
        <a
          href="/"
        >
          Home
        </a>
      </div>
      <i
        aria-hidden="true"
        class="right angle icon divider"
      />
      <div
        class="section"
      >
        <a
          href="/domain-security/123"
        />
      </div>
      <i
        aria-hidden="true"
        class="right angle icon divider"
      />
      <div
        class="active section"
      >
        SPF
      </div>
    </div>
  </div>
  <h1
    class="ui header DomainSecurityTitle--header"
  >
    SPF
  </h1>
  <div
    class="DomainSecurityTitle--icon"
  />
  <div
    class="ui hidden divider"
  />
  <p>
    bleh
  </p>
</div>
`;
